lib/database/migrations/20241127123000-create-gaq-views.js

```diff
@@ -0,0 +1,135 @@
+'use strict';
+
+const CREATE_GAQ_PERIODS_TIMESTAMPS_VIEW = `
+CREATE OR REPLACE VIEW gaq_periods_timestamps AS
+        SELECT
+            data_pass_id,
+            run_number,
+            timestamp AS \`from\`,
+            NTH_VALUE(timestamp, 2) OVER (
+                PARTITION BY data_pass_id,
+                run_number
+                ORDER BY ap.timestamp
+                ROWS BETWEEN CURRENT ROW AND 1 FOLLOWING
+            ) AS \`to\`
+        FROM (
+                -- Two selects for runs' timestamps (in case QC flag's eff. period doesn't start at run's start or end at run's end )
+                (
+                    SELECT gaqd.data_pass_id,
+                        gaqd.run_number,
+                        COALESCE(UNIX_TIMESTAMP(first_tf_timestamp), UNIX_TIMESTAMP(time_start), 0) AS timestamp
```

Comment:
I am pretty sure that all `NOW(3)` calls are evaluated to same value when called in one query.
In the  view, I used, indeed, `COALESCE(UNIX_TIMESTAMP(first_tf_timestamp), UNIX_TIMESTAMP(time_start), 0)`,
and analogously for `end timestamps`. The issue here is, to handle properly situation when all start or end timestamps of given runs are not defined, then we have nullish boundary for QC flag, but in the view for GAQ timestamps I want to have properly ordered timestamps, in particular, when we have nullish end boundary I want to have at the end of list, and when nullish start boundary then at beginning, therefore there is coalesce to `NOW(3)` and `0` respectively, because it allow such ordering

Reply:
Yes I totally understand the ordering. What I proposed is to split in two and select 2 columns, one for ordering and one that you will actually return

Reply:
applied